### DISCUSSION TITLE
nwg-shell on ArchLabs Linux since 2022.08.21 iso

### DISCUSSION THREAD

**Discussion by nwg-piotr** [nwg-piotr/nwg-shell#17] [General]
**Starting from 2022.08.21 iso, nwg-shell is no longer offered as the ArchLabs Linux sway session. Reasons aside, this has a good side: the packages will be up to date now.**

# Installation

## Alongside another WM

To install sway w/ nwg-shell alongside another window manager (I only tested it with DK), you need:

- the `base-devel` package
- web browser, text editor, file manager (by default: `chromium`, `mousepad` and `thunar`)

### Steps

1. `baph -inN nwg-shell`
2. `nwg-shell-installer -a`

## As the only window manager

This will require almost all the steps enumerated in the [How to set up nwg-shell on minimal Arch Linux install in several simple steps](https://github.com/nwg-piotr/nwg-shell/wiki) Wiki. You need to clone and install `baph`, install and update `xdg-user-dirs` and also set up [pulseaudio](https://wiki.archlinux.org/title/PulseAudio) or [pipewire](https://wiki.archlinux.org/title/PipeWire). You don't need to modify `/etc/profile` to add `~/.local/bin/` to the path (this has already been done for you), but you need to create the folder itself, as it does not exist.

The above is valid for the 2022.08.21 ArchLabs iso. Having it done, just execute steps 1 and 2 described in the previous chapter.

# Migrating pre-2022.08.21 sway session

This is possible, but quite difficult, and not really supported. Will be easier with a little help from the [pacseek](https://github.com/moson-mo/pacseek) [AUR] package.

You need to uninstall the `archlabs-nwg-shell` package, and then replace all the AUR stuff previously packaged for the ArchLabs repository (they are "local" now) with AUR counterparts:

`archlabs-autotiling` -> `autotiling`
`archlabs-azote` -> `azote`
`archlabs-gopsuinfo` -> `gopsuinfo`
`archlabs-nwg-bar` -> `nwg-bar-bin`
`archlabs-nwg-dock` -> `nwg-dock-bin`
`archlabs-nwg-drawer` -> `nwg-drawer-bin`
`archlabs-nwg-menu` -> `nwg-menu-bin`
`archlabs-nwg-look` -> `nwg-look-bin`
`archlabs-nwg-panel` -> `nwg-panel`
`archlabs-nwg-shell-config` -> `nwg-shell-config`
`archlabs-nwg-wrapper` -> `nwg-wrapper`
`archlabs-nwg-displays` -> `nwg-displays`
`archlabs-swaync` -> `swaync`
`archlabs-nwg-shell` -> `nwg-shell`

In some cases you may need to uninstall the `archlabs-` package first. For sure it affects the `archlabs-gtklock` package.

In case you get into trouble, fell free to request help here.

**Comment by Peppe61** [comment]
Hi Piotr, great you continue here anyway. I will keep on following you,  for the software but mainly bcs you're a nice guy. Want to thank you for all your great contributions, I guess this is how it goes sometimes. For me personally it is the end of a great AL journey, the added value has become minimal now. Take care,  stay in touch,  Mark (aka Peppe)

**Reply by nwg-piotr** [reply]
Thank you, Mark!

I've been the AL forum member since 2018, before I even tried sway. There's a bunch of great folks there, and I'm not going to leave this place as long as it remains friendly.

Adding nwg-shell to AL was a tempting idea, but it was a mistake. Firstly, due to AL's minimalist philosophy, and secondly because of the need to package AUR software into the AL repo. The frequency of package updates was a PITA for both natemaia and myself. Having no response to my requests for 3 weeks, I lost my patience a bit. It will be better for AL and for further shell development. Until now, I had to do a lot of things twice. Now it will be easier.

**Reply by Peppe61** [reply]
Yes,  agree,  great people there. Although I miss drummer. Anyway,  for that I will still hang around, for my distros I will move on. Cheers mate!
